File: src/main/java/common/dto/UserSubMenuDto.java
package common.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import common.model.SubMenu;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;



@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserSubMenuDto {
    private Long pageId;
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private SubMenu subMenu;
    private boolean newAuth;
    private boolean editAuth;
    private boolean deleteAuth;
    private boolean printAuth;
    private boolean approveAuth;
    private boolean cancelAuth;
    private boolean acceptAuth;
    private boolean completeAuth;
}

File: src/app/xgarage/dashboard/dto/usersubmenudto.ts

import { SubMenu } from '../model/submenu';

export interface UserSubMenuDto{
    pageId?: number;
    subMenu?: SubMenu;
    newAuth?: boolean;
    editAuth?: boolean;
    deleteAuth?: boolean;
    printAuth?: boolean;
    approveAuth?: boolean;
    cancelAuth?: boolean;
    acceptAuth?: boolean;
    completeAuth?: boolean;

}